core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -414,7 +416,13 @@ class DiplomacyManager() : IsPartOfGameInfoSerialization {
     fun updateHasOpenBorders() {
         // City-states can enter ally's territory (the opposite is true anyway even without open borders)
         val newHasOpenBorders = civInfo.getAllyCiv() == otherCivName
-                || trades.flatMap { it.theirOffers }.any { it.name == Constants.openBorders && it.duration > 0 }
+                || trades.flatMap { it.ourOffers }.any { it.name == Constants.openBorders && it.duration > 0 }
+        if (civInfo.civName == "Egypt") {
+            println("updateHasOpenBorders")
+            println(civInfo.civName)
+            println(trades.flatMap { it.theirOffers })
+            println()
+        }
```

Comment:
These changes seem to be a debug code, not related to the issue. Also, they change behaviour, please, revert.
FYI: `git add -i` and `git add -p` are really useful tools to avoid commits like this. You control what to include into commit file-by-file and line-by-line.

Reply:
Thank you for tips. I will use these commands in the future.